airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java

```diff
@@ -148,4 +154,53 @@ public static void executeSync(Supplier<Connection> connectionFactory, String qu
     };
   }
 
+  // todo (cgardens) - this only handles results without nesting. the type conversion isn't very well
+  // tested and it makes some strong assumptions about how to handle nulls that may only make sense in
+  // testing scenarios. if you are using this for a non-testing use case, this method wil need a
+  // little more scrutiny. putting it here, because it is still a better base than nothing and if it's
+  // not here it will never be found.
+  public static List<JsonNode> resultSetToJson(ResultSet resultSet) throws SQLException {
+    final List<JsonNode> jsons = new ArrayList<>();
+    final int columnCount = resultSet.getMetaData().getColumnCount();
+
+    while (resultSet.next()) {
+      final ObjectNode json = (ObjectNode) Jsons.jsonNode(new HashMap<>());
```

Comment:
could we populate the hashmap then create objnode?

Reply:
oh sure. that's a good idea.